superbench/benchmarks/micro_benchmarks/ib_validation_performance.py

```diff
@@ -0,0 +1,320 @@
+        super().add_parser_arguments()
+
+        self._parser.add_argument(
+            '--ib_index',
+            type=int,
+            default=0,
+            required=False,
+            help='The index of ib device.',
+        )
+        self._parser.add_argument(
+            '--iters',
+            type=int,
+            default=5000,
+            required=False,
+            help='The iterations of running ib command',
+        )
+        self._parser.add_argument(
+            '--msg_size',
+            type=int,
+            default=None,
+            required=False,
+            help='The message size of running ib command, e.g., 8388608.',
+        )
+        self._parser.add_argument(
+            '--commands',
+            type=str,
+            nargs='+',
+            default=['ib_write_bw'],
+            help='The ib command used to run, e.g., {}.'.format(' '.join(self.__support_ib_commands)),
+        )
+        self._parser.add_argument(
+            '--gid_index',
+            type=int,
+            default=0,
+            required=False,
+            help='Test uses GID with GID index taken from command.',
+        )
+        self._parser.add_argument(
+            '--pattern',
+            type=str,
+            default='one-to-one',
+            required=False,
+            help='Test IB traffic pattern type, e.g., {}.'.format(''.join(self.__patterns)),
+        )
+        self._parser.add_argument(
+            '--config',
+            type=str,
+            default=None,
+            required=False,
+            help='The path of config file on the target machines',
+        )
+        self._parser.add_argument(
+            '--bidirectional', action='store_true', default=False, help='Measure bidirectional bandwidth.'
+        )
+        self._parser.add_argument(
+            '--gpu_index', type=int, default=None, required=False, help='Test Use GPUDirect with the gpu index.'
+        )
+        self._parser.add_argument(
+            '--hostfile',
+            type=str,
+            default='/root/hostfile',
+            required=False,
+            help='The path of hostfile on the target machines',
+        )
+
+    def __one_to_many(self, n):
+        config = []
+        for client in range(n):
+            row = []
+            for server in range(n):
+                if server != client:
+                    pair = '{},{}'.format(server, client)
+                    row.append(pair)
+            row = ';'.join(row)
+            config.append(row)
+        return config
+
+    def __many_to_one(self, n):
+        config = []
+        for server in range(n):
+            row = []
+            for client in range(n):
+                if server != client:
+                    pair = '{},{}'.format(server, client)
+                    row.append(pair)
+            row = ';'.join(row)
+            config.append(row)
+        return config
+
+    def __roundRobin(self, candidates):
```

Comment:
First, please write the comments for this algorithm.

Reply:
I think your implementation is circle method for Round-robin tournament (https://en.wikipedia.org/wiki/Round-robin_tournament)